cmd/kubeadm/app/apis/kubeadm/v1alpha1/defaults.go

```diff
@@ -105,6 +113,35 @@ func SetDefaults_MasterConfiguration(obj *MasterConfiguration) {
 
 	SetDefaultsEtcdSelfHosted(obj)
 	SetDefaults_KubeletConfiguration(obj)
+	SetDefaults_ProxyConfiguration(obj)
+
+}
+
+// SetDefaults_ProxyConfiguration asigns default values for the Proxy
+func SetDefaults_ProxyConfiguration(obj *MasterConfiguration) {
+
+	if obj.KubeProxy.Config == nil {
+		obj.KubeProxy.Config = &kubeproxyconfigv1alpha1.KubeProxyConfiguration{}
+	}
+
+	if obj.KubeProxy.Config.BindAddress == "" {
```

Comment:
this has to move to `util/config/masterconfig.go` as `cfg.API.AdvertiseAddress` isn't populated until there using the dynamic defaults